### PR TITLE
mgr/cephadm: provision prometheus

### PR DESCRIPTION
**Pull request by liewegas**
- [x] Merge the default image PR first (see #33069)
- [ ] Include other mgr endpoints (not just active mgr)
- [ ] Expand the config file to include other items (hosts, alertmanager)

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -1549,46 +1549,72 @@ def _create_daemon(self, daemon_type, daemon_id, host,
...
+  - /etc/prometheus/alerting/*
+scrape_configs:
+  - job_name: 'ceph'
+    static_configs:
+    - targets: {mgr_scrape_list}
+""".format(
+    mgr_scrape_list=str(mgr_scrape_list)
+    )
             })
+            extra_args.extend(['--config-json', '-'])
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: This deserves it's own function

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -2015,6 +2040,56 @@ def update_rbd_mirror(self, spec):
         spec = NodeAssignment(spec=spec, get_hosts_func=self._get_hosts, service_type='rbd-mirror').load()
         return self._update_service('rbd-mirror', self.add_rbd_mirror, spec)
 
+    def add_prometheus(self, spec):
+        if not spec.placement.hosts or len(spec.placement.hosts) < spec.count:
+            raise RuntimeError("must specify at least %d hosts" % spec.count)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/module.py]: 1. This deserves it's own function: `def placement_has_hosts(placement: PlacementSpec)`
2. https://docs.ceph.com/docs/master/mgr/orchestrator_modules/#orchestrator.OrchestratorValidationError

File: src/pybind/mgr/orchestrator.py [hunk 3]
```diff
@@ -1004,6 +1004,24 @@ def update_rgw(self, spec):
...
+        # type: (str) -> Completion
+        """Remove a Prometheus service"""
+        raise NotImplementedError()
+
+    def update_prometheus(self, spec):
+        # type: (StatefulServiceSpec) -> Completion
+        """
+        Update / redeploy existing RGW zone
+        Like for example changing the number of service instances.
+        """
```
Comment (p-se) [inline src/pybind/mgr/orchestrator.py]:
```
"""Update/redeploy a Prometheus service"""
```

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 4]
```diff
@@ -581,6 +581,46 @@ def _nfs_rm(self, svc_id):
...
+        self._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        return HandleCommandResult(stdout=completion.result_str())
+
+    @orchestrator._cli_write_command(
+        'orchestrator prom update',
+        'name=num,type=CephInt,req=false '
+        'name=hosts,type=CephString,n=N,req=false '
+        'name=label,type=CephString,req=false',
+        'Create a prometheus service')
```
Comment (p-se) [inline src/pybind/mgr/orchestrator_cli/module.py]: `'Update a prometheus service')`

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
Comment (liewegas) [reply src/pybind/mgr/cephadm/module.py]: @jschmid1 do we need this at all with the scheduler?

**Pull request merged**